### ISSUE TITLE
The memory of Primary process keep increasing

### ISSUE THREAD

**Issue by weickdev** [arangodb/arangodb#5257]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [ ] 3.2
- [x] 3.3
- [ ] self-compiled devel branch

Mode:
- [x] Cluster
- [x] Single-Server

Storage-Engine:
- [ ] mmfiles
- [x] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [x] other: CentOS7
- [ ] Windows, version: 
- [ ] MacOS, version:

I want to control the total memory the ADB uses, so I set below parameters:
```
--cache.size 1073741824 \
--rocksdb.block-cache-size 1073741824 \
--javascript.v8-max-heap 1024
```
If I run ADB in single-server mode, it works. The single process uses about only 3G memory. But if I run ADB in cluster mode, the setting seems useless. The primary db process uses 7G memory（90% of my total memory) and keep increasing.

How I can control the total memory in cluster mode?

**Comment by weickdev** [comment]
I try to insert 5000000 documents, and then recurrently insert the same 5000000 documents, the memory keeps increasing too.

**Comment by dothebart** [comment]
As discussed in https://github.com/arangodb/arangodb/issues/11577 [exposing more options of rocksdb to the commandline](https://www.arangodb.com/docs/stable/programs-arangod-options.html#rocksdb) seems to ease the memory situation.

Namely:
 -   `--rocksdb.cache-index-and-filter-blocks-with-high-priority`
 -   `--rocksdb.pin-l0-filter-and-index-blocks-in-cache`
 -   `--rocksdb.pin-top-level-index-and-filter`

These have been added to [the now released ArangoDB 3.7](https://www.arangodb.com/2020/08/arangodb-3-7-a-big-step-forward-for-multi-model/) - hence closing this.
